# see https://github.com/nextui-org/nextui/issues/1954
diff --git a/dist/chunk-QXNXBY53.mjs b/dist/chunk-QXNXBY53.mjs
index 90cc2cacd05cf9c4e986d08589bd4875308f4b7c..5793e00cbe16e565dcf0442576b544f94cc65e3c 100644
--- a/dist/chunk-QXNXBY53.mjs
+++ b/dist/chunk-QXNXBY53.mjs
@@ -13,6 +13,7 @@ import { mergeRefs } from "@nextui-org/react-utils";
 import { createDOMRef } from "@nextui-org/react-utils";
 import { useMemo, useRef, useCallback } from "react";
 import { toReactAriaPlacement, getArrowPlacement } from "@nextui-org/aria-utils";
+import { useSafeLayoutEffect } from "@nextui-org/use-safe-layout-effect";
 function useTooltip(originalProps) {
   const [props, variantProps] = mapPropsVariants(originalProps, popover.variantKeys);
   const {
@@ -96,7 +97,7 @@ function useTooltip(originalProps) {
     shouldFlip,
     containerPadding
   });
-  useLayoutEffect(() => {
+  useSafeLayoutEffect(() => {
     if (!updatePositionDeps.length)
       return;
     updatePosition();
diff --git a/dist/use-tooltip.js b/dist/use-tooltip.js
index 0ec6e7495fd935cc975c179a6ae7feae536eb744..21bff941611cfddfb1bfdcacaa1a7e2ecf8b171d 100644
--- a/dist/use-tooltip.js
+++ b/dist/use-tooltip.js
@@ -36,6 +36,7 @@ var import_react_utils = require("@nextui-org/react-utils");
 var import_react_utils2 = require("@nextui-org/react-utils");
 var import_react2 = require("react");
 var import_aria_utils = require("@nextui-org/aria-utils");
+var import_use_safe_layout_effect = require("@nextui-org/use-safe-layout-effect");
 function useTooltip(originalProps) {
   const [props, variantProps] = (0, import_system.mapPropsVariants)(originalProps, import_theme.popover.variantKeys);
   const {
@@ -119,7 +120,7 @@ function useTooltip(originalProps) {
     shouldFlip,
     containerPadding
   });
-  (0, import_react.useLayoutEffect)(() => {
+  (0, import_use_safe_layout_effect.useSafeLayoutEffect)(() => {
     if (!updatePositionDeps.length)
       return;
     updatePosition();
