# see https://github.com/TanStack/query/issues/5458
diff --git a/build/legacy/useBaseQuery.cjs b/build/legacy/useBaseQuery.cjs
index 9be499fcb1a34d8e32675ff3c38e051d8621884f..937162589e5a30912f9fa4e10b82f3622102728f 100644
--- a/build/legacy/useBaseQuery.cjs
+++ b/build/legacy/useBaseQuery.cjs
@@ -76,6 +76,7 @@ function useBaseQuery(options, Observer, queryClient) {
       defaultedOptions
     )
   );
+  observer.setClient(client);
   const result = observer.getOptimisticResult(defaultedOptions);
   const shouldSubscribe = !isRestoring && options.subscribed !== false;
   React.useSyncExternalStore(
diff --git a/build/legacy/useBaseQuery.js b/build/legacy/useBaseQuery.js
index dccb42412e4d9a730fcf601f49a42f759d311908..ce744d4823ec1e81343a2e1c62762f456aa9cb58 100644
--- a/build/legacy/useBaseQuery.js
+++ b/build/legacy/useBaseQuery.js
@@ -52,6 +52,7 @@ function useBaseQuery(options, Observer, queryClient) {
       defaultedOptions
     )
   );
+  observer.setClient(client)
   const result = observer.getOptimisticResult(defaultedOptions);
   const shouldSubscribe = !isRestoring && options.subscribed !== false;
   React.useSyncExternalStore(
diff --git a/build/modern/useBaseQuery.cjs b/build/modern/useBaseQuery.cjs
index 864d4bb1c473c366b34f07cf12cc7c88bb4cdb37..07f914e4b787c6e68c20121f23e9f4eaae53209f 100644
--- a/build/modern/useBaseQuery.cjs
+++ b/build/modern/useBaseQuery.cjs
@@ -74,6 +74,7 @@ function useBaseQuery(options, Observer, queryClient) {
       defaultedOptions
     )
   );
+  observer.setClient(client)
   const result = observer.getOptimisticResult(defaultedOptions);
   const shouldSubscribe = !isRestoring && options.subscribed !== false;
   React.useSyncExternalStore(
diff --git a/build/modern/useBaseQuery.js b/build/modern/useBaseQuery.js
index 187d0257c39034a244c82da9dc4450dc40b90965..68c67cecd93f71e37096f00d1befbbf9ee405005 100644
--- a/build/modern/useBaseQuery.js
+++ b/build/modern/useBaseQuery.js
@@ -50,6 +50,7 @@ function useBaseQuery(options, Observer, queryClient) {
       defaultedOptions
     )
   );
+  observer.setClient(client)
   const result = observer.getOptimisticResult(defaultedOptions);
   const shouldSubscribe = !isRestoring && options.subscribed !== false;
   React.useSyncExternalStore(
diff --git a/src/useBaseQuery.ts b/src/useBaseQuery.ts
index 06690b544f4ec66ca48890a8d28f91b0e96f2943..13d44f3694a5374dcdc0b274660d4bd28a723dc6 100644
--- a/src/useBaseQuery.ts
+++ b/src/useBaseQuery.ts
@@ -89,6 +89,7 @@ export function useBaseQuery<
       ),
   )
 
+  observer.setClient(client)
   // note: this must be called before useSyncExternalStore
   const result = observer.getOptimisticResult(defaultedOptions)
 
