diff --git a/react.d.ts b/react.d.ts
index 8646581f052daf191521a143b5f9dd9e9440693b..aedd8a37f1a1c0ef972f98d465cb9b53556a1f91 100644
--- a/react.d.ts
+++ b/react.d.ts
@@ -1,9 +1,9 @@
 import type { Mutate, StateCreator, StoreApi, StoreMutatorIdentifier } from './vanilla';
-type ExtractState<S> = S extends {
+export type ExtractState<S> = S extends {
     getState: () => infer T;
 } ? T : never;
 type ReadonlyStoreApi<T> = Pick<StoreApi<T>, 'getState' | 'subscribe'>;
-type WithReact<S extends ReadonlyStoreApi<unknown>> = S & {
+export type WithReact<S extends ReadonlyStoreApi<unknown>> = S & {
     getServerState?: () => ExtractState<S>;
 };
 export declare function useStore<S extends WithReact<StoreApi<unknown>>>(api: S): ExtractState<S>;
